jms/src/main/scala/akka/stream/alpakka/jms/impl/JmsConnector.scala

```diff
@@ -56,26 +57,40 @@ private[jms] trait JmsConnector[S <: JmsSession] {
   override def preStart(): Unit = {
     // keep two elements since the time between initializing and connected can be very short.
     // always drops the old state, and keeps the most current (two) state(s) in the queue.
-    val pair = Source.queue[InternalConnectionState](2, OverflowStrategy.dropHead).preMaterialize()(this.materializer)
-    connectionStateQueue = pair._1
-    connectionStateSourcePromise.complete(Success(pair._2))
+    val (queue, source) =
+      Source
+        .queue[InternalConnectionState](2, OverflowStrategy.dropHead)
```

Comment:
IIUC, the behaviour of the queue (`connectionStateQueue`) is the same as before because it has its own overflow strategy.